src/assembler.jl

```diff
@@ -64,3 +64,122 @@ function assemble!{T}(g::AbstractVector{T}, ge::AbstractVector{T}, edof::Abstrac
         g[edof[i]] += ge[i]
     end
 end
+
+immutable AssemblerSparsityPattern{Tv, Ti}
+    K::SparseMatrixCSC{Tv, Ti}
+    f::Vector{Tv}
+    tmpi::Vector{Int}
+    tmpf::Vector{Tv}
+end
+
+
+function start_assemble(K::SparseMatrixCSC, f::Vector=Float64[])
+    fill!(K.nzval, 0.0)
+    AssemblerSparsityPattern(K, f, Int[], eltype(K)[])
+end
+
+assemble!(A::AssemblerSparsityPattern, Ke::AbstractMatrix, dofs::AbstractVector{Int}) = assemble!(A, eltype(Ke)[], Ke, dofs)
+function assemble!(A::AssemblerSparsityPattern, fe::AbstractVector, Ke::AbstractMatrix, dofs::AbstractVector{Int})
```

Comment:
This should have time complexity `O(nd * nz) + O(nd * log(nd))` where nd is the number of dofs and nz is the number of non zeros per column. The first term is the insertion and the second is the sorting.

Just inserting by binary search would have complexity `O(nd^2 * log(nz))` because we insert `nd^2` values into sorted lists with `nz` values. For large `nz` the former method should be slower. We could benchmark what the threshold is and switch depending on the lengths.